redbot/core/bot.py

```diff
@@ -886,7 +886,13 @@ async def ignored_channel_or_guild(
             if ctx.channel.type is not discord.ChannelType.private:
                 raise TypeError("Can't check permissions for non-private PartialMessageable.")
             is_private = True
-        perms = ctx.channel.permissions_for(author)
+        if isinstance(ctx, discord.Interaction):
+            ctx: discord.Interaction
+            perms = ctx.permissions
+            # The permissions here are always linked to the interaction author &
+            # the channel it originated from.
+        else:
+            perms = ctx.channel.permissions_for(author)
```

Comment:
`Context` can also be based on an interaction:
```suggestion
        if isinstance(ctx, discord.Message):
            perms = ctx.channel.permissions_for(author)
        else:
            # `permissions` attribute will use permissions from the interaction when possible,
            # or `ctx.channel.permissions_for(author)` for non-interaction contexts.
        	perms = ctx.permissions
```